### PR TITLE
stratisd encryption blackbox test

### PR DESCRIPTION
**Pull request by jbaublitz**
Related to #510

### CODE REVIEW

**Comment by jbaublitz** [comment]
@mulkieran This is ready for initial review. Do we also want a pool destroy test for encrypted volumes?

File: tests/blackbox/testlib/utils.py [hunk 1]
```diff
@@ -123,3 +125,67 @@ def exec_test_command(cmd):
...
+                "Command '%s' failed with exit code %s" % (" ".join(args), return_code)
+            )
+
+    def __enter__(self):
+        with open("/dev/urandom", "rb") as urandom_f:
+            key_desc = base64.b64encode(urandom_f.read(16)).decode("utf-8")
+
+        args = ["keyctl", "get_persistent", "@s", "0"]
+        exit_values = run(args, capture_output=True, text=True)
+        persistent_id = exit_values.stdout.strip()
```
Comment (mulkieran) [inline tests/blackbox/testlib/utils.py:170]: Please swap these two lines; dosn't make sense to get ```persistent_id``` if call has failed.

File: tests/blackbox/testlib/utils.py [hunk 2]
```diff
@@ -123,3 +125,67 @@ def exec_test_command(cmd):
...
+        :param args: The command line that caused the command to fail
+        :type args: list of str
+        :raises RuntimeError
+        """
+        if return_code != 0:
+            raise RuntimeError(
+                "Command '%s' failed with exit code %s" % (" ".join(args), return_code)
+            )
+
+    def __enter__(self):
```
Comment (mulkieran) [inline tests/blackbox/testlib/utils.py:164]: Please add header docs here, especially about returned value.

File: tests/blackbox/testlib/utils.py (hunk 1 above)
Comment (mulkieran) [reply tests/blackbox/testlib/utils.py:170]: Actually, do you even need ```persistent_id```? man page indicates that you could use ```@s```.

Comment (jbaublitz) [reply tests/blackbox/testlib/utils.py:170]: I'll do a test to see if this is necessary, but this comes down to how we'd recommend users administer the keychain. I'll present both options once I have more information, but the way this is currently operating, it does work so the only benefit we will be getting out of this is a slightly simpler set of commands to recommend (which may still be worth it).

Comment (jbaublitz) [reply tests/blackbox/testlib/utils.py:170]: It appears we do need `persistent_id`.

**Comment by jbaublitz** [comment]
Blocked by stratis-storage/stratisd#1961

File: tests/blackbox/testlib/utils.py (hunk 1 above)
Comment (jbaublitz) [reply tests/blackbox/testlib/utils.py]: Resolved.

File: tests/blackbox/testlib/utils.py (hunk 2 above)
Comment (jbaublitz) [reply tests/blackbox/testlib/utils.py:164]: Resolved.

**Comment by bgurney-rh** [comment]
I tried another blackbox test run with these commits after stratis-storage/stratisd#1961 was merged, and everything passes.

**Comment by jbaublitz** [comment]
@bgurney-rh Excellent! I was hoping it would match up with my local testing.

**Pull request closed without merging**